test/allowlist_for_publicAPI.json

```diff
@@ -1987,7 +1987,7 @@
     "DimConstraints",
     "DimDynamic",
     "DynamicDimConstraintPrinter",
-    "EqualityConstraint",
+    "EqualityConstraints",
```

Comment:
Adding or updating entries in this file is not ok. Please fix the public API of the module that you are using as suggested in the error message from test_public_bindings!

Reply:
There is no error message from test_public_bindings (ran `buck2 test '@fbcode//mode/opt' fbcode//caffe2/test:public_bindings` to confirm). I was renaming `EqualityConstraint` to `EqualityConstraints` throughout. In any case this isn't really supposed to be a public API.

Reply:
@albanD I could make `EqualityConstraints` private by adding `_`. However there seem to be many other things like it in this file we could also make private. cc @ezyang 

I prefer consistency of naming, so I'm torn between leaving this as is and making more things private. (Of course I could just do this for `EqualityConstraints` and move on, but I feel this is more process than needed in this PR.)

Reply:
A PR either making them all nicely public or nicely private would be good yes!

Reply:
OK to leave this as is in this PR, then?

Reply:
It's best if you can do that other PR first and then rebase this on top since the general policy is that this allowlist is remove-only.

Reply:
Eh, in that case I'd rather just back out the renaming and land this PR first.